src/net_processing.cpp

```diff
@@ -129,8 +129,6 @@ static constexpr double BLOCK_DOWNLOAD_TIMEOUT_BASE = 1;
 static constexpr double BLOCK_DOWNLOAD_TIMEOUT_PER_PEER = 0.5;
 /** Maximum number of headers to announce when relaying blocks with headers message.*/
 static const unsigned int MAX_BLOCKS_TO_ANNOUNCE = 8;
-/** Maximum number of unconnecting headers announcements before DoS score */
-static const int MAX_NUM_UNCONNECTING_HEADERS_MSGS = 10;
```

Comment:
The commit message of a903bba4d65720265659c6d4dae6cd8d9ec89c0f says that https://github.com/bitcoin/bitcoin/pull/25454 has largely addressed the problem of a request cycle, but I don't understand why:
If we receive an unconnecting header, we send a `GetHeaders`. But if our peer is broken in a way that they ignore the contents of our locator and send us unconnecting headers again, we clear `m_last_getheaders_timestamp`, call `ProcessHeadersMessage` and immediately send a `GetHeaders` again (we wold only delay that if `m_last_getheaders_timestamp` hadn't been cleared)  - and after this PR we wouldn't discourage the peer anymore so that this could go on forever. How did #25454 address that?

Reply:
Glad you caught this @mzumsande.